web-server/src/components/Settings/SyncDaysSetting.tsx

```diff
@@ -0,0 +1,146 @@
+import { FlexBox } from '../FlexBox';
+import { Box, Button, Divider, TextField } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import { Line } from '../Text';
+import { FC, useEffect, useState } from 'react';
+import { useAuth } from '@/hooks/useAuth';
+import { useSnackbar } from 'notistack';
+import { useDispatch, useSelector } from '@/store';
+import {
+  getDefaultSyncDaysSettings,
+  updateDefaultSyncDaysSettings
+} from '@/slices/org';
+import { FetchState } from '@/constants/ui-states';
+
+export const SyncDaysSetting: FC = () => {
+  const dispatch = useDispatch();
+  const { orgId } = useAuth();
+  const { enqueueSnackbar } = useSnackbar();
+  const initialSyncDays = useSelector((state) => state.org.defaultSyncDays);
+  const [syncDays, setSyncDays] = useState<number>(initialSyncDays);
+  const isSaving = useSelector(
+    (s) => s.org.requests?.defaultSyncDays === FetchState.REQUEST
+  );
+  const [error, setError] = useState('');
+
+  const getSyncDays = async (): Promise<number> => {
+    const res = await dispatch(getDefaultSyncDaysSettings({ orgId }));
+    return res.payload as number;
+  };
+
+  const updateSyncDays = async (days: number): Promise<number> => {
+    const res = await dispatch(
+      updateDefaultSyncDaysSettings({ orgId, defaultSyncDays: days })
+    );
+    return res.payload as number;
+  };
+
+  const handleDiscard = () => {
+    setSyncDays(initialSyncDays);
+    setError('');
+  };
+
+  const handleSave = async () => {
+    try {
+      await updateSyncDays(syncDays);
+      enqueueSnackbar('Default Sync Days Updated Successfully', {
+        variant: 'success',
+        autoHideDuration: 3000
+      });
+    } catch (e) {
+      handleDiscard();
+      return enqueueSnackbar('Default Sync Days Update Failed', {
```

Comment:
If API request fails, we will set Input value to previous Sync days and Show error.

Since updateSyncDays(syncDays) will not throw any error as Redux handles it internally, we can either use then and catch to explicitly throw error [Reference](https://github.com/middlewarehq/middleware/pull/516/files/345fda0a8f57a83f24c81a5b4292b93cc89cfa74#r1723346414).

Or

handle using requestStatus whether request is fulfilled or rejected. I think then even try catch will not be needed.
```js
const handleSave = async () => {
    try {
      const res = await dispatch(
        updateDefaultSyncDaysSettings({ orgId, defaultSyncDays: syncDays })
      ); // This never throws error
      
      if (res.meta.requestStatus === 'rejected') {
        throw new Error();
      }
      enqueueSnackbar('Default Sync Days Updated Successfully', {
        variant: 'success',
        autoHideDuration: 3000
      });
    } catch (e) {
      handleDiscard();
      return enqueueSnackbar('Default Sync Days Update Failed', {
        variant: 'error',
        autoHideDuration: 2000
      });
    }
  };
```

Reply:
oh, cool!
Just wrap it in a `useCallback`

``` typescript
const func = useCallback(async () => {}, [])
```

Reply:
@e-for-eshaan If sync days setting is updated successfully we need to call /sync on sync server.

This can be done by calling Nextjs internal API which does force sync, making a entry in UIPreferences table.
```js
handleApi(`/internal/${orgId}/sync_repos`, { method: 'POST' });
```

But I feel instead of force sync, it should be just a normal sync call. But we can't directly call flask sync server API.

Or treating this as force sync is fine?

Reply:
@Kamlesh72 (unrelated to your last comment)
After the sync setting is updated successfully, I'd recommend making the sync_repos API call on the node server (a.k.a. BFF, a.k.a. nextjs API) instead from the browser.

That way you save time on another request round-trip from the browser to the server.

Reply:
@Kamlesh72 why cant we call `syncReposForOrg` after the defauilt sync days setting updation returns a 200 ?

Reply:
This func calls /sync

Reply:
<img width="1440" alt="Screenshot 2024-08-21 at 6 37 43 PM" src="https://github.com/user-attachments/assets/155dd0c8-68d4-4989-93ab-029af3bf6593">


CHATGPT: The error you're encountering is likely because the Next.js /api routes are server-side only, and the fs module is also server-side (it’s a Node.js module). When you try to call syncReposForOrg from within a React component (which runs on the client side), it attempts to use the fs module, leading to the error.

To resolve this issue, you should ensure that the syncReposForOrg function is only used in server-side contexts or modify it to make a client-side API call instead of directly using the fs module.

Reply:
`syncReposForOrg` is a server side only function, and should be kept there.
you can call it in any of your API pages